src/include/btree_inline.h

```diff
@@ -1651,14 +1653,22 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_splitp)
     page = ref->page;
     mod = page->modify;
 
-    /* A truncated page can't be evicted until the truncate completes. */
-    if (__wt_page_del_active(session, ref, true))
-        return (false);
-
-    /* Otherwise, never modified pages can always be evicted. */
+    /* Never modified pages can always be evicted. */
     if (mod == NULL)
         return (true);
 
+    /*
+     * Don't attempt to evict fast-truncate pages until the fast-truncate transaction completes. We
+     * are only here if eviction takes an interest in the fast-truncate page, which implies the page
+     * was instantiated. If the fast-truncate transaction wasn't yet resolved at instantiation, an
+     * update list was created for the transaction to use on commit or abort, and then removed when
+     * the transaction was resolved, so it's our flag.
```

Comment:
A bit of long comment which while understandable I found a bit confusing, I had a go at re-writing it let me know what you think:
```
/*
 * Don't attempt to evict fast-truncate pages until the fast-truncate transaction is committed or rolled-back
 * as the location of updates associated with it are held on the ref. A non null update chain implies that this
 * page was previously truncated and re-instantiated. Therefore it's ref state cannot be deleted. Once the truncate
 * is resolved the update array is cleared.
 */
```

Reply:
That chunk of code existed in a bunch of different variants during development, I'm not surprised the comment is confusing. I've pushed a new version in 1aa029169, please let me know if you think it's sufficient.